extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -244,23 +329,30 @@ class AmpApesterMedia extends AMP.BaseElement {
           };
           this.iframe_ = iframe;
           this.element.appendChild(overflow);
-          this.element.appendChild(iframe);
-          return this.iframePromise_ = this.loadPromise(iframe).then(() => {
+
+          this.registerToApesterEvents_();
+
+          return this.iframePromise_ =
+        this.getMediaContent(media.content, src).then(content => {
+          return installFriendlyIframeEmbed(this.iframe_, this.element, {
```

Comment:
we can't allow your dynamic content to be in a friendly iframe which has access to the top-level page context. Only reviewed code can access top-level context. /cc @dvoytenko to confirm my statement.

Reply:
Ho.. We are getting back the HTML for the media back from our servers.
No way to leverage it on the AMP pages?

Reply:
Not outside of a 3P iframe. We can't allow your HTML access to the 1P page (FriendlyIframe is same origin and has access to the page). 

What you like to do is certainly possible but you would need to use our 3P frame approach (amp-facebook, amp-twitter, etc.. do this already). In this approach, we will create an iframe with a different origin for you and your specified JS file runs inside that frame, it can fetch whatever and replace the content as needed. Communication between the component and the frame still need to be through post messages.